src/test/java/com/github/nylle/javafixture/InstanceFactoryTest.java

```diff
@@ -103,6 +134,16 @@ void factoryMethodWithArgument() {
 
             assertThat(result.getValue()).isNotNull();
         }
+        @Test
+        @DisplayName("optional can be created (and may be empty)")
+        void createOptional() {
+            var sut = new InstanceFactory(new SpecimenFactory(new Context(Configuration.configure())));
+
+            var result = sut.manufacture(new SpecimenType<Optional<String>>(){});
+
+            assertThat(result).isInstanceOf(Optional.class);
+            assertThat(result.orElse("optional may be empty")).isInstanceOf(String.class);
```

Comment:
Nice!

Reply:
Thank you. I think it hides the
```java
if ( ! result.isEmpty())
  assertThatItContainsString
```
pretty well. Seriously, I could not find a better way to assert that it may randomly be either empty or contain a string.